src/js/common/containers/DowntimeNotification.jsx

```diff
@@ -0,0 +1,234 @@
+
+      // Remove null values (services that have no known downtime)
+      .filter(downtime => !!downtime)
+
+      // Put each value into the corresponding status of the downtimeMap
+      .reduce((map, downtime) => {
+        const status = this.getStatusForDowntime(downtime, now, nextHour);
+        return {
+          ...map,
+          [status]: map[status].concat(downtime)
+        };
+      }, downtimeMap);
+  }
+
+  determineStatus(downtimeMap) {
+    if (this.props.determineStatus) return this.props.determineStatus(downtimeMap);
+
+    const statusDown = downtimeMap[serviceStatus.down].length > 0;
+    const statusDownApproaching = downtimeMap[serviceStatus.downtimeApproaching].length > 0;
+    const statusOk = !statusDown && !statusDownApproaching;
+
+    if (statusOk) return serviceStatus.ok;
+    if (statusDown) return serviceStatus.down;
+
+    return serviceStatus.downtimeApproaching;
+  }
+
+  renderStatusDown({ endTime }) {
+    const title = `The ${this.props.appTitle} is down for maintenance.`;
+    let message = <p>We’re making some updates to {this.props.appTitle}. We’re sorry it’s not working right now. Please check back soon.</p>;
+    if (endTime) {
+      message = (
+        <p>We’re making some updates to {this.props.appTitle}. We’re sorry it’s not working right now, and we hope to be finished by {endTime.format('MMMM Mo, LT')}. Please check back soon.</p>
+      );
+    }
+
+    const downtimeNotification = <div><h2>{title}</h2>{message}</div>;
+
+    return (
+      <DowntimeNotificationWrapper status={serviceStatus.down}>
+        {downtimeNotification}
+      </DowntimeNotificationWrapper>
+    );
+  }
+
+  renderStatusDownApproaching({ startTime, endTime }) {
+    const title = `The ${this.props.appTitle} will be down for maintenance soon`;
+    const message = <p>We'll be doing some work on {this.props.appTitle} on {startTime.format('MMMM Mo')} between {startTime.format('LT')} and {endTime.format('LT')} . If you have trouble using this tool during that time, please check back soon.</p>;
+    let downtimeNotification = null;
+    if (!this.state.modalDismissed) {
+      const close = () => this.setState({ modalDismissed: true });
+      downtimeNotification = (
+        <Modal id="downtime-approaching-modal"
+          title={title}
+          onClose={close}
+          visible={!this.state.modalDismissed}>
+          {message}
+          <button type="button" className="usa-button-secondary" onClick={close}>Dismiss</button>
+        </Modal>);
+    }
+    return (
+      <DowntimeNotificationWrapper status={serviceStatus.downtimeApproaching}>
+        {downtimeNotification}
+        {this.props.children || this.props.content}
+      </DowntimeNotificationWrapper>
+    );
+  }
+
+  render() {
+    if (!this.props.isReady) return this.props.loadingIndicator || <LoadingIndicator message={`Checking ${this.props.appTitle} status...`}/>;
```

Comment:
This would be clearer, I think, if it weren't a one line if statement.

Reply:
Yeah good call. At one point it was nice and clean as a one liner but not so much now.